### PR TITLE
Minor improvements to SpotBugs and Dependency Check parsers

### PR DESCRIPTION
**Pull request by adiffpirate**
**SpotBugs**

The report contains some HTML code that was not being properly parsed.
Now, using BeautifulSoup, we can work with tags and deliver a more precise report.
Enhancements:
- A more clear mitigation.
- References moved to where they belong with links that work.

dd_spotbugs_compare

**Dependency Check**
- Added mitigation

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Thanks. Are there any unit tests / samples you could add that "prove" the new html parsing part is working correctly?

File: dojo/tools/spotbugs/parser.py [hunk 1]
```diff
@@ -4,9 +4,8 @@
 __status__ = "Development"
 
 import re
-
+from bs4 import BeautifulSoup
```
Comment (damiencarol) [inline dojo/tools/spotbugs/parser.py:7]: We use `html2text` for this part of the project. Please switch to the package `html2text`

**Review by damiencarol — CHANGES_REQUESTED** [review]
Can't introduce BS instead of `html2text` that we use in a lot of parsers

**Comment by adiffpirate** [comment]
I'm not sure if these unittests for the HTML parsing are good enough. Let me know what you guys think ;)
_(Also I'm sorry for taking so long to update this PR, it's been a crazy couple of months)_

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        'jsonlines==2.0.0',  # requred by yarn audit parser
+    ],
+
+    extras_require={'mysql': ['mysqlclient==2.0.3']},
+
+    dependency_links=[
+        "https://github.com/grendel513/python-pdfkit/tarball/master#egg=pdfkit-0.5.0"
+    ],
+    url='https://github.com/DefectDojo/django-DefectDojo'
+)
```
Comment (valentijnscholten) [inline setup.py:75]: this file was removed

File: dojo/tools/dependency_check/parser.py [hunk 3]
```diff
@@ -238,6 +238,10 @@ def get_finding_from_vulnerability(self, dependency, related_dependency, vulnera
                                      'source: {1}\n' \
                                      'url: {2}\n\n'.format(name, source, url)
 
+        component_name, component_version = self.get_component_name_and_version_from_dependency(dependency, related_dependency, namespace)
+
```
Comment (valentijnscholten) [inline dojo/tools/dependency_check/parser.py:242]: don't think this is needed/ intended?

**Review by valentijnscholten — CHANGES_REQUESTED** [review]

File: dojo/tools/spotbugs/parser.py (hunk 1 above)
Comment (adiffpirate) [reply dojo/tools/spotbugs/parser.py]: Done :)

File: dojo/tools/dependency_check/parser.py (hunk 3 above)
Comment (adiffpirate) [reply dojo/tools/dependency_check/parser.py]: Yup, it was not intended, thanks for catching that

File: setup.py (hunk 2 above)
Comment (adiffpirate) [reply setup.py]: Oops, sorry for that

**Review by valentijnscholten — APPROVED** [review]

**Review by damiencarol — APPROVED** [review]

**Pull request merged**